### PR TITLE
refactor: move payment system icons into its own token

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

File: projects/addon-commerce/components/input-card/input-card.options.ts [hunk 1]
```diff
@@ -18,22 +19,10 @@ export interface TuiInputCardOptions {
...
-        options,
-        TUI_INPUT_CARD_DEFAULT_OPTIONS,
-    );
-}
+export const TUI_INPUT_CARD_OPTIONS: InjectionToken<TuiInputCardOptions> =
+    tuiCreateTokenFromFactory(() => ({
+        cardSrc: '',
+        autocompleteEnabled: false,
+        paymentSystemHandler: tuiGetPaymentSystem,
+        icons: inject(TUI_PAYMENT_SYSTEM_ICONS_OPTIONS),
```
Comment (waterplea) [inline projects/addon-commerce/components/input-card/input-card.options.ts:27]: No point in duplicating it in another token

Comment (splincode) [reply projects/addon-commerce/components/input-card/input-card.options.ts]: fixed

File: projects/addon-commerce/components/input-card/abstract-input-card.ts [hunk 2]
```diff
@@ -18,6 +18,8 @@ export abstract class AbstractTuiInputCard<
     extends AbstractTuiNullableControl<T>
     implements TuiFocusableElementAccessor
 {
+    protected abstract paymentIcons: Record<TuiPaymentSystem, string>;
```
Comment (waterplea) [inline projects/addon-commerce/components/input-card/abstract-input-card.ts:21]: Why not inject it here?

File: projects/addon-commerce/components/thumbnail-card/thumbnail-card.component.ts [hunk 3]
```diff
@@ -35,6 +36,7 @@ export class TuiThumbnailCardComponent {
     public iconRight = '';
 
     protected readonly options = inject(TUI_INPUT_CARD_OPTIONS);
```
Comment (waterplea) [inline projects/addon-commerce/components/thumbnail-card/thumbnail-card.component.ts:38]: Do we still need it here?

**Review by waterplea — APPROVED** [review]

File: projects/addon-commerce/components/input-card/abstract-input-card.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-commerce/components/input-card/abstract-input-card.ts]: fixed

File: projects/addon-commerce/components/thumbnail-card/thumbnail-card.component.ts (hunk 3 above)
Comment (splincode) [reply projects/addon-commerce/components/thumbnail-card/thumbnail-card.component.ts]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**